tensordict/tensordict.py

```diff
@@ -959,10 +959,16 @@ def __ne__(self, other: object) -> TensorDictBase:
             tensors of the same shape as the original tensors.
 
         """
-        if not isinstance(other, (TensorDictBase, dict, float, int)):
-            return False
+        from tensordict.prototype import is_tensorclass
+
+        if not isinstance(
+            other, (TensorDictBase, dict, float, int, torch.Tensor, MemmapTensor)
+        ) and not is_tensorclass(other):
+            return True
         if not isinstance(other, TensorDictBase) and isinstance(other, dict):
```

Comment:
I would probably switch the order:
- tensordictbase
- dict
- tensorclass
- numeric + tensor
- other (aka return True) 

reading "positive" ifs is easier than all these negations